symengine/number.cpp

```diff
@@ -5,6 +5,11 @@
 namespace SymEngine
 {
 
+RCP<const Basic> Number::conjugate() const
+{
+    throw NotImplementedError("Not Implemented.");
```

Comment:
Instead of throwing an error, you can first check that the number is not complex, and return a pointer to self if it is not complex. Then you don't have to override the method in many classes.